packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -670,11 +673,11 @@ class ClipRRectLayer extends ContainerLayer {
   void addToScene(ui.SceneBuilder builder, Offset layerOffset) {
     bool enabled = true;
     assert(() {
-      enabled = !debugDisableClipLayers;
+      enabled = !debugDisableClipLayers && clip != Clip.none;
```

Comment:
this code only runs in debug mode. Is that what you intended?

Reply:
Ah, no. Changed.